File: bench/python/simd_map2.py
# SIMD cross-language benchmark: elementwise (a[i] + b[i]). Matches bench/simd_map2.march
import time

n = 5_000_000
a = [(i % 100) / 100.0 for i in range(n)]
b = [(i % 100) / 100.0 + 1.0 for i in range(n)]
t0 = time.perf_counter()
combined = [x + y for x, y in zip(a, b)]
t1 = time.perf_counter()
total = 0.0
for x in combined:
    total += x
print(f"RESULT {total}")
print(f"TIME_MS {(t1 - t0) * 1000.0}")

File: bench/rust/simd_map2.rs
// SIMD cross-language benchmark: elementwise (a[i] + b[i]). Matches bench/simd_map2.march
use std::time::Instant;

fn main() {
    let n: usize = 5_000_000;
    let a: Vec<f64> = (0..n).map(|i| (i % 100) as f64 / 100.0).collect();
    let b: Vec<f64> = (0..n).map(|i| (i % 100) as f64 / 100.0 + 1.0).collect();
    let t0 = Instant::now();
    let combined: Vec<f64> = a.iter().zip(b.iter()).map(|(x, y)| x + y).collect();
    let elapsed = t0.elapsed();
    let total: f64 = combined.iter().sum();
    println!("RESULT {}", total);
    println!("TIME_MS {}", elapsed.as_secs_f64() * 1000.0);
}
